### PR TITLE
DOCS-1721: Add build CLI commands

### PR DESCRIPTION
**Pull request by andf-viam**
Add brief reference docs for `module build` subcommands
- Including additional `meta.json` object `build` to add
- Including Zack's awesome id --> logs workflow as well.

Questions:
- Do users have to set up cloud runner? Is there billing? Or is `module build start` all I need to get on Viam-provided hardware with my hefty build job?

### CODE REVIEW

**Comment by andf-viam** [comment]
Hi @zaporter-work , @abe-winter . Here's a tiny quick update to add reference entries for the new `module build` CLI commands. Left a few questions in the PR description, apologies for my ignorance on how we cloud build. Thanks much, happy to make any changes!

**Comment by abe-winter** [comment]
- Do users have to set up cloud runner?
  - not sure what this one is asking
- Is there billing?
  - no, but there are rate limits
- Or is module build start all I need to get on Viam-provided hardware with my hefty build job?
  - our runners are fairly small so idk about 'hefty', but yes

File: docs/fleet/cli.md [hunk 1]
```diff
@@ -686,6 +695,57 @@ See [Upload a custom module](/registry/upload/#upload-a-custom-module) and [Upda
 
 See [Modular resources](/registry/) for a conceptual overview of modules and the modular resource system at Viam.
 
+##### Using the `build` subcommand
+
+You can use the `module build start` or `module build local` commands to build your custom module according to the build steps you specify in your <file>meta.json</file> file.
+
+You could use `build start` to build or compile your module on a cloud build host that might offer additional platform support than you have access to locally, or you could use `build local` to quickly test that your module builds or compiles as expected on your local hardware.
```
Comment (sguequierre) [inline docs/fleet/cli.md:702]:
```suggestion
Use `build start` to build or compile your module on a cloud build host that might offer additional platform support than you have access to locally, or use `build local` to quickly test that your module builds or compiles as expected on your local hardware.
```
Think the conditional here is a little off, would reword to be more direct

File: docs/fleet/cli.md [hunk 2]
```diff
@@ -686,6 +695,57 @@ See [Upload a custom module](/registry/upload/#upload-a-custom-module) and [Upda
...
+    "build": "make module.tar.gz",
+    "path": "module.tar.gz",
+    "arch": ["linux/amd64", "linux/arm64"]
+  },
+  "entrypoint": "run.sh"
+}
+```
+
+When you initiate a build job using either `start` or `local`, the command returns the build ID of your job.
+You can then provide that build ID to the `module build logs` command to show the relevant build logs for that build.
```
Comment (sguequierre) [inline docs/fleet/cli.md:741]:
```suggestion
Provide that build ID to the `module build logs` command to show the relevant build logs for that build.
```
Conditional

File: docs/fleet/cli.md [hunk 3]
```diff
@@ -686,6 +695,57 @@ See [Upload a custom module](/registry/upload/#upload-a-custom-module) and [Upda
...
+
+For example, you might use the following to initiate a build, and return the build logs as soon as it completes:
```
Comment (sguequierre) [inline docs/fleet/cli.md:743]:
```suggestion
For example, use the following to initiate a build, and return the build logs as soon as it completes:
```

**Review by sguequierre — APPROVED** [review]
LGTM minor phrasing comments

File: docs/fleet/cli.md (hunk 2 above)
Comment (zaporter-work) [inline docs/fleet/cli.md:741]: We should mention the `module build list` command here. It will show the build they just started and the status for all the builds

**Review by zaporter-work — APPROVED** [review]
LGTM!
